contrib/node/src/python/pants/contrib/node/subsystems/node_distribution.py

```diff
@@ -233,3 +274,36 @@ def yarnpkg_command(self, args, node_paths=None):
     """
     return self._command_gen(
       [self.install_node, self.install_yarnpkg], 'yarnpkg', args=args, node_paths=node_paths)
+
+  def _configure_eslinter(self, bootstrapped_support_path):
+    logger.debug('Copying {setupdir} to bootstrapped dir: {support_path}'
+                           .format(setupdir=self._eslint_setupdir, support_path=bootstrapped_support_path))
+    safe_rmtree(bootstrapped_support_path)
+    shutil.copytree(self._eslint_setupdir, bootstrapped_support_path)
+    return True
+
+  def fetch_eslint_supportdir(self):
+    """ Returns the path where the ESLint is bootstrapped.
+
+    :returns: The path where ESLint is bootstrapped and whether or not it already installed
+    :rtype: (string, bool)
+    """
+    bootstrap_dir = os.path.realpath(os.path.expanduser(self._pants_bootstrapdir))
+    # TODO(nsaechao): ./pants clean-all should handle removing this directory.
+    bootstrapped_support_path = os.path.join(bootstrap_dir, self._eslint_supportdir)
+
+    configured = False
+    if self._eslint_setupdir:
+      configured = self._binary_util.is_bin_valid(self._eslint_setupdir,
+                                           [BinaryUtil.BinaryFileSpec('package.json'),
+                                            BinaryUtil.BinaryFileSpec('yarn.lock')])
+    if not configured:
+      safe_mkdir(bootstrapped_support_path, clean=True)
+    else:
+      binary_file_specs = [
+        BinaryUtil.BinaryFileSpec(f, hash_file(os.path.join(self._eslint_setupdir, f)))
+        for f in ['yarn.lock', 'package.json']]
+      installed = self._binary_util.is_bin_valid(bootstrapped_support_path, binary_file_specs)
+      if not installed:
+        self._configure_eslinter(bootstrapped_support_path)
+    return (bootstrapped_support_path, configured)
```

Comment:
I found the conditionals here a bit hard to follow. From what I can tell, these are the cases.
- If setupdir isn't set, or if it doesn't contain `package.json`, `yarn.lock`, then create/clean the bootstrap path, returning `bootstrap, False`
- if setupdir contains the files but bootstrap doesn't, or bootstrap's are different, then run configure_eslinter and return `bootstrap, True`
- if setupdir contains the files and they are the same as the ones in bootstrap, then return `bootstrap, True`

Does that sound right?